packages/core/src/util/stackTrace.js

```diff
@@ -37,11 +37,17 @@ exports.captureStackTrace = function captureStackTrace(length, referenceFunction
     // drawback is that it might show too much, but that's better than not having any stack trace at all.
     Error.captureStackTrace(stackTraceTarget);
   }
-  const stack = stackTraceTarget.stack;
+
+  let stack = stackTraceTarget.stack;
+
+  if (!Array.isArray(stack)) {
+    stack = [];
+  }
+
   Error.stackTraceLimit = originalLimit;
   Error.prepareStackTrace = originalPrepareStackTrace;
 
-  if (drop > 0) {
+  if (drop > 0 && stack.length >= drop) {
```

Comment:
IMO there were two bugs.

1. stack not being a string (the actual use case is not 100% known yet)
2. stack.length >= drop behaviour

Reply:
I wonder when (2) would have happened in practice (without the new stack = [] fix)

Reply:
I am not sure tbh, I just looked for potential bugs in this fn :)